applications/Chat/coati/models/chatglm/modeling_chatglm.py

```diff
@@ -0,0 +1,1439 @@
+"""
+This code is copied from https://huggingface.co/THUDM/chatglm-6b/resolve/main/modeling_chatglm.py
+"""
+
+""" PyTorch ChatGLM model. """
+
```

Comment:
In order to ensure compatibility with the existing codebase, we need to implement three functionalities within the ChatGLMActor:

- When a pre-trained model path is supplied, initialize the model using pre-trained parameters.

- Should a custom model configuration be available, create an instance of the ChatGLM model using the provided custom configuration.

- In the absence of either of the aforementioned options, instantiate the ChatGLM model using the default ChatGLMConfig.

Hence, we need the definitions for both ChatGLMConfig and the ChatGLM model.  Since the ChatGLM doesn't expose them, it's necessary to replicate and include the definition code within our repository.